### PR TITLE
Refactoring/43 set mandatory properties

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java [hunk 1]
```diff
@@ -38,6 +41,12 @@ public AbstractSqlDialect(final ConnectionFactory connectionFactory, final Adapt
         this.properties = properties;
     }
 
+    protected static List<String> addAdditionalSupportedProperties(final List<String> additionalPropertiesToSupport) {
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java:44]: Add JavaDoc for this protected method please.

Comment (redcatbear) [inline src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java:44]: The method does not do what the name says. It has a side effect.

File: src/test/java/com/exasol/adapter/dialects/derby/DerbySqlDialect.java [hunk 2]
```diff
@@ -19,9 +20,11 @@
 public class DerbySqlDialect extends AbstractSqlDialect {
     static final String NAME = "DERBY";
     private static final Capabilities CAPABILITIES = createCapabilityList();
-    private static final List<String> SUPPORTED_PROPERTIES = Arrays.asList(SQL_DIALECT_PROPERTY,
-            CONNECTION_NAME_PROPERTY, CATALOG_NAME_PROPERTY, SCHEMA_NAME_PROPERTY, TABLE_FILTER_PROPERTY,
-            EXCLUDED_CAPABILITIES_PROPERTY, DEBUG_ADDRESS_PROPERTY, LOG_LEVEL_PROPERTY);
+    private static final List<String> SUPPORTED_PROPERTIES = createSupportedPropertiesList();
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/dialects/derby/DerbySqlDialect.java:23]: Make this an instance field and pull it up to the base class.
Remove side effect from `add...`.
Don't override `getCapabilities` anymore.

File: src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java]: done

Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/dialects/AbstractSqlDialect.java]: fixed

File: src/test/java/com/exasol/adapter/dialects/derby/DerbySqlDialect.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/derby/DerbySqlDialect.java]: fixed

**Pull request merged**